ra/registration-authority.go

```diff
@@ -602,6 +608,12 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(ra.clk.Now()) {
+		err = fmt.Errorf("Expired authorization")
```

Comment:
Okay, talked to Jacob and he mentioned you say that the WFE is doing this. The problem is there's a race between the WFE check and the check in the RA and this one could still get caught.